server/publications/collections/products.js

```diff
@@ -165,33 +128,48 @@ function filterProducts(productFilters) {
       }, {
         slug: { $in: shopIdsOrSlugs }
       }]
+    }, {
+      fields: {
+        _id: 1
+      }
     }).map((shop) => shop._id);
 
-    // If we found shops, update the productFilters
-    if (shopIds) {
-      productFilters.shops = shopIds;
-    } else {
-      return false;
-    }
+    activeShopIds = _.intersection(activeShopIds, shopIds);
+  }
+
+  if (activeShopIds.length) {
+    return {
+      ...selector,
+      shopId: { $in: activeShopIds }
+    };
+  }
+
+  return selector;
+}
+
+function filterProducts(productFilters) {
+  // if there are filter/params that don't match the schema
+  // validate, catch except but return no results
+  try {
+    if (productFilters) filters.validate(productFilters);
+  } catch (e) {
```

Comment:
Do we make an exception re: single letter variables here?

Reply:
shouldn't be, looks like it slipped through code review. Nice catch.

Reply:
Looks like it came in through a community contribution, still shouldn't have slipped through code review though. I'll fix it.

Reply:
I would have swore that ESLint caught those in the past